staging/src/k8s.io/apiserver/pkg/cel/openapi/resolver/definitions.go

```diff
@@ -68,6 +69,22 @@ func (d *DefinitionsSchemaResolver) ResolveSchema(gvk schema.GroupVersionKind) (
 			return nil, false
 		}
 		s := def.Schema
+
+		//!TODO: Fix codegen to do this
+		//!TODO: I think this bug also affects VAP? int-or-string fields
+		// have no type and are ignored by CEL decltype construction
+		if len(s.Type) == 0 && len(s.OneOf) == 2 && len(s.OneOf[0].Type) > 0 && len(s.OneOf[1].Type) > 0 {
```

Comment:
Just to double check, do we have examples of this outside of the "Quantity" object? We can probably easily fix the generation for quantity, does this also affect custom declared types that may also not be setting `int-or-string` properly?

Reply:
Edit: I see another example for the `IntOrString` type. Two approaches I see:
1. Allow users to define another method/marker to indicate that a type is an int or string. This would be in addition to the `OpenAPIV3OneOfTypes()` method.
2. Hack the generator code to perform this same check. Unfortunately we cannot evaluate `OpenAPIV3OneOfTypes()` statically codegen and it would need to be done at runtime.

Reply:
Have a prototype of 2 here: https://github.com/kubernetes/kube-openapi/pull/478